cli/js/compiler_imports.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+import {
+  MediaType,
+  SourceFile,
+  SourceFileJson
+} from "./compiler_sourcefile.ts";
+import { cwd } from "./dir.ts";
+import * as dispatch from "./dispatch.ts";
+import { sendAsync, sendSync } from "./dispatch_json.ts";
+import { assert } from "./util.ts";
+import * as util from "./util.ts";
+
+/** Resolve a path to the final path segment passed. */
+function resolvePath(...pathSegments: string[]): string {
+  let resolvedPath = "";
+  let resolvedAbsolute = false;
+
+  for (let i = pathSegments.length - 1; i >= -1 && !resolvedAbsolute; i--) {
+    let path: string;
+
+    if (i >= 0) path = pathSegments[i];
+    else path = cwd();
+
+    // Skip empty entries
+    if (path.length === 0) {
+      continue;
+    }
+
+    resolvedPath = `${path}/${resolvedPath}`;
+    resolvedAbsolute = path.charCodeAt(0) === util.CHAR_FORWARD_SLASH;
+  }
+
+  // At this point the path should be resolved to a full absolute path, but
+  // handle relative paths to be safe (might happen when cwd() fails)
+
+  // Normalize the path
+  resolvedPath = util.normalizeString(
+    resolvedPath,
+    !resolvedAbsolute,
+    "/",
+    code => code === util.CHAR_FORWARD_SLASH
+  );
+
+  if (resolvedAbsolute) {
+    if (resolvedPath.length > 0) return `/${resolvedPath}`;
+    else return "/";
+  } else if (resolvedPath.length > 0) return resolvedPath;
+  else return ".";
+}
+
+/** Resolve a relative specifier based on the referrer.  Used when resolving
+ * modules internally within the runtime compiler API. */
+function resolveSpecifier(specifier: string, referrer: string): string {
+  if (!specifier.startsWith(".")) {
+    return specifier;
+  }
+  const pathParts = referrer.split("/");
+  pathParts.pop();
+  let path = pathParts.join("/");
+  path = path.endsWith("/") ? path : `${path}/`;
+  return resolvePath(path, specifier);
+}
+
+/** Ops to Rust to resolve modules' URLs. */
+export function resolveModules(
+  specifiers: string[],
+  referrer?: string
+): string[] {
+  util.log("compiler_imports::resolveModules", { specifiers, referrer });
+  return sendSync(dispatch.OP_RESOLVE_MODULES, { specifiers, referrer });
```

Comment:
Looks like this fixed #3054

Reply:
It would.  Forgot we opened an issue for it.

Reply:
Added it to the top note.